arkouda/strings.py

```diff
@@ -1473,6 +1473,98 @@ def lstick(self, other: Strings, delimiter: Union[bytes, str_scalars] = "") -> S
     def __radd__(self, other: Strings) -> Strings:
         return self.lstick(other)
 
+    def get_prefixes(
+        self, n: int_scalars, return_origins: bool = True, proper: bool = True
+    ) -> Union[Strings, Tuple[Strings, pdarray]]:
+        """
+        Return the n-long prefix of each string, where possible
+
+        Parameters
+        ----------
+        n : int
+            Length of prefix
+        return_origins : bool
+            If True, return a logical index indicating which strings
+            were long enough to return an n-prefix
+        proper : bool
+            If True, only return proper prefixes, i.e. from strings
+            that are at least n+1 long. If False, allow the entire
+            string to be returned as a prefix.
+
+        Returns
+        -------
+        prefixes : Strings
+            The array of n-character prefixes; the number of elements is the number of
+            True values in the returned mask.
+        origin_indices : pdarray, bool
+            Boolean array that is True where the string was long enough to return
+            an n-character prefix, False otherwise.
+        """
+        repMsg = cast(str, generic_msg(
+            cmd="segmentedSubstring",
+            args={
+                "objType": self.objtype,
+                "name": self.name,
+                "nChars": n,
+                "returnOrigins": return_origins,
+                "kind": "prefixes",
+                "proper": str(proper).lower(),
```

Comment:
I think you should be able to just pass `self` and `proper` here. Ethans message processing code should take care of the rest. So
```python
 args={
      "objType": self.objtype,
      "name": self,
      "nChars": n,
      "returnOrigins": return_origins,
      "kind": "prefixes",
      "proper": proper,
}
```

Reply:
Thanks, this is cool!